acid/features/history/tests/test_model.py

```diff
@@ -12,49 +12,129 @@
         start.__get__ = mocker.Mock(return_value=None)
         buildset = make_buildset()
         assert buildset.duration is None
 
+    @db_session
     def test_duration_wo_end_should_return_null(self, make_buildset, mocker):
         end = mocker.patch.object(ZuulBuildSet, 'end_datetime')
         end.__get__ = mocker.Mock(return_value=None)
         buildset = make_buildset()
         assert buildset.duration is None
 
+    @db_session
     def test_build_number_should_return_int(self, make_buildset):
         buildset = make_buildset(build_number=104)
         assert buildset.build_number == 104
 
+    @db_session
+    def test_build_number_return_none_if_no_build_number(self, make_buildset):
+        buildset = make_buildset(build_number=None)
+        assert buildset.build_number is None
+
+    @db_session
+    def test_get_branches_return_branches(self, make_buildset):
+        make_buildset(branch='master')
+        make_buildset(branch='gimp')
+        branches = list(ZuulBuildSet.get_branches())
+        expected = ['master', 'gimp']
+        assert branches == expected
+
+    @db_session
+    def test_get_for_pipeline_return_pipelines(self,
+                                               database_with_buildsets):
+        database = database_with_buildsets(count=2)
+        database[0].pipeline = "one"
+        return_buildsets = list(ZuulBuildSet.get_for_pipeline(pipeline='one'))
+        expected = [database[0]]
+        assert return_buildsets == expected
+
+    @db_session
+    def test_get_filtered_if_branch_is_not_in_branches(self,
+                                                       database_with_buildsets):
+        database = database_with_buildsets()
+        for buildset in database:
+            buildset.ref = 'refs/heads/gimp'
+        filtered_buildsets = list(ZuulBuildSet.get_filtered(
+            pipeline='periodic-nightly',
+            branch='master',
+            build='105'))
+        expected = [database[0]]
+        assert filtered_buildsets == expected
+
+    @db_session
+    def test_get_filtered_if_build_is_none(self,
+                                           database_with_buildsets):
+        buildsets = database_with_buildsets()
+        filtered_buildsets = list(ZuulBuildSet.get_filtered(
+            pipeline='periodic-nightly',
+            branch='master',
+            build=None))
+        expected = buildsets[::-1]
+        assert filtered_buildsets == expected
+
+    @pytest.mark.parametrize("start_time, end_time ,expected", [
+        (0, 0, timedelta(0, 6900)),
+        (0, None, None),
+        (None, 0, None),
+        (None, None, None)
+    ])
+    @db_session
+    def test_duration_return_value(self, make_buildset, start_time, end_time,
+                                   expected):
+        buildset = make_buildset(number_of_builds=1, start_time=start_time,
+                                 end_time=end_time)
+        builds_duration = [b.duration for b in buildset.builds][0]
```

Comment:
what is the purpose of iterating over the buildset if you use only first element? Wouldn't `buildset.builds[0].duration` be simpler?